examples/solvers/pdhg_denoising_L2_HuberTV.py

```diff
@@ -0,0 +1,139 @@
+"""Linearly convergent total variation denoising using PDHG.
+
+This exhaustive example solves the L2-HuberTV problem
+
+        min_{x >= 0}  1/2 ||x - d||_2^2
+            + lam * sum_i eta_gamma(||grad(x)_i||_2)
+
+where grad the spatial gradient and d is given noisy data. Here eta_gamma
+denotes the Huber function. For more details, see the Huber documentation.
+
+We compare two different step size rules as described in
+
+Chambolle, A., & Pock, T. (2011). *A First-Order Primal-Dual Algorithm for
+Convex Problems with Applications to Imaging. Journal of Mathematical Imaging
+and Vision, 40(1), 120–145. http://doi.org/10.1007/s10851-010-0251-1
+
+Chambolle, A., Ehrhardt, M. J., Richtárik, P., & Schönlieb, C.-B. (2017).
+Stochastic Primal-Dual Hybrid Gradient Algorithm with Arbitrary Sampling and
+Imaging Applications. Retrieved from http://arxiv.org/abs/1706.04957
+
+and show their convergence rates.
+
+For further details and a description of the solution method used, see
+https://odlgroup.github.io/odl/guide/pdhg_guide.html in the ODL documentation.
+"""
+
+import numpy as np
+import scipy
+import odl
+import matplotlib.pyplot as plt
+
+# --- define setting --- #
+
+# Define ground truth, space and noisy data
+image = np.rot90(scipy.misc.ascent()[::2, ::2].astype('float'), 3)
+shape = image.shape
+image /= image.max()
+space = odl.uniform_discr([0, 0], shape, shape)
+orig = space.element(image.copy())
+d = odl.phantom.white_noise(space, orig, 0.1)
+
+# Define objective functional
+op = odl.Gradient(space)  # operator
+op.norm = np.sqrt(8) + 1e-4  # norm with forward differences is well-known
```

Comment:
We shouldn't be setting attributes on instances like this. In fact, this overrides the *method* `Operator.norm` with a constant.
Until `Gradient.norm` is implemented please make this a free variable instead. Do we have an open issue on that somewhere? I can't find one.

Reply:
OK, changed.